File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        Set<ListNode> data = new HashSet<>();
        while(headA!=null){
            data.add(headA);
            headA = headA.next;
        }
        while(headB!=null){
            if(data.contains(headB)){
                return headB;
            }
            headB = headB.next;
        }
        return null;
    }
}

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        set<ListNode*> mySet;
        while(headA != nullptr){
            mySet.insert(headA);
            headA = headA->next;
        }
        while(headB != nullptr){
            if(mySet.count(headB) > 0){
                return headB;
            }
            headB = headB->next;
        }
        return nullptr;
    }
};